docker-compose/1_Auto_Upstall/noco.sh

```diff
@@ -172,6 +173,41 @@ prompt() {
     fi
 }
 
+prompt_oneof() {
+    local response
+    local resp_upper
+    local oneof_text
+    local prompt_text="$1"
+    local default_response="$2"
+    shift 1
+
+    prompt_text+=" (default: $default_response) "
+    oneof_text+="["
+    for one in "$@"; do
+        oneof_text+="$one,"
+    done
+    oneof_text="${oneof_text%%,}]"
+    prompt_text+="${oneof_text}: "
+
+    while true; do
+        read -r -p "$prompt_text" response
+        [ -z "$response" ] && echo "$default_response"
+
+        for one in "$@"; do
+            resp_upper="${response^^}"
+
+            if [ "$response" = "$one" ]; then
+                echo "$one"
+                return
+            elif [ "$resp_upper" = "$one" ]; then
+                echo "$one"
+                return
+            fi
```

Comment:
we can convert both values to upper case and then compare, this way we can avoid this duplicate code

```bash
            resp_upper="${response^^}"
            one_upper="${one^^}"

            if [ "$resp_upper" = "$one_upper" ]; then
                echo "$one"
                return
            fi
```

Reply:
thanks